sqlmesh/core/model/kind.py

```diff
@@ -153,53 +153,66 @@ def data_hash_values(self) -> t.List[t.Optional[str]]:
 
 
 class TimeColumn(PydanticModel):
-    column: str
+    column: exp.Expression
     format: t.Optional[str] = None
 
     @classmethod
     def validator(cls) -> classmethod:
         def _time_column_validator(v: t.Any) -> TimeColumn:
             if isinstance(v, exp.Tuple):
-                kwargs = {
-                    key: v.expressions[i].name
-                    for i, key in enumerate(("column", "format")[: len(v.expressions)])
-                }
-                return TimeColumn(**kwargs)
+                column_expr = v.expressions[0]
+                return TimeColumn(
+                    column=(
+                        exp.column(column_expr)
+                        if isinstance(column_expr, exp.Identifier)
```

Comment:
you tested this? you get back an identifier here?

Reply:
I tested this and, yes, there are cases when I get an identifier there.